kit/Kit.cpp

```diff
@@ -1355,6 +1404,12 @@ void Document::handleSaveMessage(const std::string &)
     // if a bgsave process - now we can clean up.
     if (_isBgSaveProcess)
     {
+        LOG_TRC("BgSave completed");
+        if (BgSaveWatchdog)
+        {
+            BgSaveWatchdog->complete();
```

Comment:
Almost certainly we want to join that thread to clean it up - but ... lets get this in for now.

Reply:
I intentionally didn't want to complicate things. First, to get a patch out sooner and with less clutter. But also because the point of joining would very likely be also the point of calling `_Exit()`.